logger.go

```diff
@@ -51,10 +57,22 @@ func (l *Logger) SetWriter(w io.Writer) {
 	l.mu.Unlock()
 }
 
+func (l *Logger) SetFlags(flags int32) {
+	l.mu.Lock()
+	l.flags = l.flags | flags
```

Comment:
not sure if would be better  to set all the  flags at once (a simple assignment) ... maybe adding a AddFlags method for doing the OR ...  What do you think?